tests/analysis/test_speed.py

```diff
@@ -0,0 +1,102 @@
+import os
+import pytest
+import trackintel as ti
+import numpy as np
+from geopandas.testing import assert_geodataframe_equal
+
+
+@pytest.fixture
+def example_triplegs():
+    """Generate input data for trip generation from geolife positionfixes"""
+    pfs, _ = ti.io.dataset_reader.read_geolife(os.path.join("tests", "data", "geolife_long"))
+    pfs, sp = pfs.as_positionfixes.generate_staypoints(
+        method="sliding", dist_threshold=25, time_threshold=5, gap_threshold=1e6
+    )
+    sp = sp.as_staypoints.create_activity_flag(time_threshold=15)
+    pfs, tpls = pfs.as_positionfixes.generate_triplegs(sp)
+    return pfs, sp, tpls
+
+
+@pytest.fixture
+def load_positionfixes():
+    """Load test positionfixes"""
+    pfs = ti.io.file.read_positionfixes_csv(os.path.join("tests", "data", "positionfixes.csv"), sep=";", index_col="id")
+    correct_speeds = np.array([8.82100607, 8.82100607, 0.36585538, 1.93127652, 19.60643425, 2.07086017])
```

Comment:
Do you calculate this using a calculator or e.g., QGIS? Could you document it somewhere? But this is a good comparison:)

Reply:
I just computed it with python and then double checked it on several values. I'm always unsure what is the best method.. I haven't worked with qgis so that would be a huge overhead for me.